src/components/routers/SwitchRouter.tsx

```diff
@@ -226,6 +226,7 @@ export class SwitchRouterForm extends NodeRouterForm<SwitchRouter, SwitchRouterS
             var cases = update(this.state.cases, { $splice: [[idx, 1]] });
             this.setState({ cases: cases });
         }
+        this.props.removeWidget(c.props.name);
```

Comment:
Fix for when removing cases. The main widget list needs to be kept up to date since it's used on validation and submission.